### PR TITLE
Exercise8

### PR DESCRIPTION
**Pull request by Karrot96**

### CODE REVIEW

File: travis/deploy_to_heroku.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+set -ev
+if [[ "$TRAVIS_BRANCH" == "master" ]]; then
+    if [[ "$TRAVIS_PULL_REQUEST" == "false" ]]; then
+        docker pull karrot96/todo-app:latest
+        docker tag karrot96/todo-app:latest registry.heroku.com/karro96-todo-app/web
+        heroku container:login
+        docker push registry.heroku.com/karro96-todo-app/web
+        heroku container:release web --app karro96-todo-app
+    fi
```
Comment (hugh-emerson) [inline travis/deploy_to_heroku.sh]: Correct me if I'm wrong, but I don't think this script will work on Travis.

- You haven't installed the heroku CLI
- You should log in with a docker command instead with something like `echo $HEROKU_API_KEY | docker login --username ... --password-stdin registry.heroku.com`. You've not logged into heroku so `heroku container:login` won't work.

**Review by hugh-emerson — CHANGES_REQUESTED** [review]
Mostly there, but see my comment on `deploy_to_heroku.sh`.

File: travis/deploy_to_heroku.sh (hunk 1 above)
Comment (Karrot96) [reply travis/deploy_to_heroku.sh]: I do believe you are wrong. I have tested this quite thoroughly and it seems to work. The command `heroku container:login` works as long as you have the environment variable `HEROKU_API_KEY` set to your auth key. The auth key can be generated from the command line and then added as a secret in the travis script.

Comment (Karrot96) [reply travis/deploy_to_heroku.sh]: There does not seem to be a requirement to have logged into heroku separately,

Comment (Karrot96) [reply travis/deploy_to_heroku.sh]: The commit `2cab237` used this script as is and published to my heroku instance without any issues as further proof of this working.

**Review by hugh-emerson — APPROVED** [review]
Discussed over Slack, but reposting here for posterity.

This build is evidence that travis-ci comes with Heroku CLI installed: https://travis-ci.com/github/Karrot96/DevOps-Course-Starter/jobs/478259285

So everything does work.

An extra point I made was that you shouldn't publish a new `:latest` image whenever any branch builds. But apart from that this looks good and should be approved in Aptem once it's resubmitted.

**Pull request merged**